rules/manager.go

```diff
@@ -459,65 +459,40 @@ func (c ruleDependencyController) AnalyseRules(rules []Rule) {
 type RuleConcurrencyController interface {
 	// Allow determines whether any concurrent evaluation slots are available.
 	// If Allow() returns true, then Done() must be called to release the acquired slot.
-	Allow() bool
+	Allow(ctx context.Context, group *Group, rule Rule) bool
```

Comment:
We need the group here for a new functionality that I'm about to introduce but would rather just modify the interface in one go.

Reply:
Is there an issue for this new functionality.

Reply:
Do we need to update the doc for this interface? 

> // RuleConcurrencyController controls concurrency for rules that are safe to be evaluated concurrently.
// Its purpose is to bound the amount of concurrency in rule evaluations to avoid overwhelming the Prometheus
// server with additional query load. Concurrency is controlled globally, not on a per-group basis.

Now it will start receiving the rule and the group to make the decision.

Reply:
Good shout, done in https://github.com/prometheus/prometheus/pull/14491/commits/ee9200590f8e7b6d3d099cae536ad804f24c94dd